### ISSUE TITLE
Debugger opens wrong <node internals> file using legacy protocol

### ISSUE THREAD

**Issue by ezolenko** [Microsoft/vscode#24298]
- VSCode Version: 1.11.1 (works in 1.10.2)
- OS Version: Win10
- Setup: an electron app that has one node subprocess launched with debug port. VS configured to attach to that port, with protocol explicitly set to "legacy". The app (screeps private server) loads multiple user scripts (one of them is being debugged). The scripts are all called `main.js` and somehow loaded through lots of other js.

VSCode 1.10.2 attaches and breaks on `debugger;` statement, retrieving correct js file from node. When adding a breakpoint VSCode shows it as `main \1000\<node internals>`. Stack trace and locals show up correctly.

VSCode 1.11.1 attaches and breaks on `debugger;` statement, retrieving code for wrong `main`. Stack trace and locals still show up correctly and I can step through with correct effects. Newly added breakpoint is shown as `main <node internals>` (missing /1000/ part comparing to 1.10.2).

This looks like a regression in 1.11.1, I reverted back to 1.10.2 and right file is opened.

**Comment by chrmarti** [comment]
Could you link/attach a simple example that reproduces the problem?

**Comment by weinand** [comment]
@isidorn is this a problem with VS Code's document caching?

**Comment by ezolenko** [comment]
Unfortunately I don't know app internals well enough to reproduce the environment outside of it. And getting the screeps server itself requires a steam purchase. 

Let me know if there is any tracing or diagnostics I can do on my side.

In case somebody has the game already here are setup steps:

- get the game on steam <http://store.steampowered.com/app/464350/>
- setup private server: <https://github.com/screeps/screeps/blob/master/README.md> (will need steam API key)
- clone included bot (simplebot) so you have at least two of them (<see https://github.com/screeps/screeps/blob/master/README.md#npc-bots>)
- install debug mod <https://github.com/ScreepsMods/screepsmod-user-script-debug>
- add `debugger;` statement into `main.js` of one of the bots (and something else to easily distinguish between the two)
- use `launch.json` from debug mod readme to attach VSCode to the server

Expected: VSCode retrieves bot's source code and pauses on `debugger` statement in the right copy.
Actual: code is from the wrong bot (if you get the right one, try breakpointing the other one).

**Comment by isidorn** [comment]
@weinand yes this is potentially a duplicate of https://github.com/Microsoft/vscode/issues/22741
To verify that it is a duplicate @ezolenko do you first correctly stop in a file (internal module) and everything is fine regarding the content. Only on subsequent debug sessions you would get the wrong content.
You should also be able to reproduce this with `protocol` set to `inspector`

**Comment by ezolenko** [comment]
No, I was getting dumped directly into the wrong file, on first attach/hit bp, on reconnect/hit bp, and on continue/re-hitting bp (because loop returns there). 

The stacktrace and variables in debugger sidebar are always seemingly correct. 

I only tried attaching though, since there is no good way to do launch from debugger in my case.

Can't test with inspector because electron doesn't support it: <https://github.com/electron/electron/issues/6634>

**Comment by isidorn** [comment]
It is strange that you always get the wrong file, even on the very first debug start when we have no stale internal modules. If that is true than I think it is not a duplicate @weinand

**Comment by ezolenko** [comment]
I don't know if that applies to #22741, but in this case confused scripts are all named `main`, so that likely has something to do with it.

**Comment by chrmarti** [comment]
@ezolenko Could you attach/post your launch.json?

**Comment by ezolenko** [comment]
```
		{
			"type": "node",
			"request": "attach",
			"name": "Attach to Process",
			"port": 6001,
			"outFiles": ["C:/Users/Evgeniy/AppData/Local/Screeps/scripts/127_0_0_1___21025/default/*.js"],
			"sourceMaps": true,
			"smartStep": false,
			"protocol": "legacy"
		}

```

**Comment by ezolenko** [comment]
This is how it looks, in case there is something useful I'm missing.

This is in build 1.11.2

![image](https://cloud.githubusercontent.com/assets/1617834/25034739/f187d476-20a5-11e7-8f19-7480b6d1223b.png)

**Comment by weinand** [comment]
@ezolenko you can hover of the name of the file in the CALL STACK view to see the full path. Do you see different paths for the various 'main' scripts?

**Comment by ezolenko** [comment]
While reproducing it again I noticed something that kinda ties back to #22741:

I set `debugger` breakpoint in main.js, line 2258. This is where freshly attached VS dropped me off:
![image](https://cloud.githubusercontent.com/assets/1617834/25064909/7c7281b8-21c2-11e7-8315-afacb8e481b7.png)

If I hit continue, this is what happens:
![image](https://cloud.githubusercontent.com/assets/1617834/25064936/d0fa189a-21c2-11e7-8061-fbb5ab3f549e.png)

So here first time breakpoint comes out in the wrong location entirely (I don't have any soft breakpoints set), but the file loaded is correct for the stack.

Second time around (when loop returns to `debugger` statement), the stack is correct, but the file loaded is still wrong (and vs runs out of lines, so highlights the last line). 

In both cases file names are `<node_internals>/main` in call stack window, and `<node_internals>\main` in the text editor.

![image](https://cloud.githubusercontent.com/assets/1617834/25064968/6af3051a-21c3-11e7-8e0b-d31ab4d02069.png)

**Comment by ezolenko** [comment]
Ah, ignore the the first part actually. I did have a rogue `debugger;` statement set on line 24 (that happened to be out of the main loop, so was hit only once after reload).

**Comment by ezolenko** [comment]
This seems to be fixed in 1.12.1. Unless it randomly started picking up code from a different vm...
